File: server-rails-single/app/policies/label_policy.rb
# frozen_string_literal: true

# Single-tenant LabelPolicy.
#
# Labels are a SHARED GLOBAL catalog (no owner, no LabelScope), so every
# authenticated user can read and manage the shared set. Permissive on
# attributes; action-level checks pass via the permissive-policy fallback.
class LabelPolicy < Rhino::ResourcePolicy
  def permitted_attributes_for_show(_user) = ["*"]
  def hidden_attributes_for_show(_user) = []
  def permitted_attributes_for_create(_user) = %w[name color]
  def permitted_attributes_for_update(_user) = %w[name color]
end

File: server-laravel-single/app/Policies/LabelPolicy.php
<?php

namespace App\Policies;

use Rhino\Policies\ResourcePolicy;
use Illuminate\Contracts\Auth\Authenticatable;

/**
 * Single-tenant LabelPolicy.
 *
 * Labels are a SHARED GLOBAL catalog (no scope). Any authenticated user may read
 * and manage the shared catalog. All attributes are exposed.
 */
class LabelPolicy extends ResourcePolicy
{
    public function permittedAttributesForShow(?Authenticatable $user): array
    {
        return $user ? ['*'] : [];
    }

    public function permittedAttributesForCreate(?Authenticatable $user): array
    {
        return $user ? ['name', 'color'] : [];
    }

    public function permittedAttributesForUpdate(?Authenticatable $user): array
    {
        return $this->permittedAttributesForCreate($user);
    }
}
